### PR TITLE
Expose `window_extent` to device ref

### PR DESCRIPTION
**Pull request by PointKernel**
This is the first step to migrate shared memory code from the legacy map to the new map. It exposes the `window_extent` ref API thus it can be used to construct container device ref on shared memory.

### CODE REVIEW

File: include/cuco/detail/static_map/static_map_ref.inl [hunk 1]
```diff
@@ -115,6 +115,26 @@ static_map_ref<Key, T, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>
...
+                                             T,
+                                             Scope,
+                                             KeyEqual,
+                                             ProbingScheme,
+                                             StorageRef,
+                                             Operators...>::extent_type
+static_map_ref<Key, T, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>::window_extent()
+  const noexcept
+{
+  return impl_.window_extent();
```
Comment (sleeepyjack) [inline include/cuco/detail/static_map/static_map_ref.inl:135]: This returns a `window_extent` which can't be implicitly converted to `size_type`. Do we want to return a `cuco::extent` instead? On second thought, maybe it's a better choice to be consistent with the return type. What do you think?

**Review by sleeepyjack — COMMENTED** [review]
Happy 400th issue/PR 🥳

File: include/cuco/detail/static_map/static_map_ref.inl (hunk 1 above)
Comment (PointKernel) [reply include/cuco/detail/static_map/static_map_ref.inl:135]: The return value of this function will be used to construct a container ref thus window extent is exactly what we need.

Comment (sleeepyjack) [reply include/cuco/detail/static_map/static_map_ref.inl:135]: Ah, right. Good point. Approved!

**Review by sleeepyjack — APPROVED** [review]

**Pull request merged**